prefetch.bs

```diff
@@ -415,3 +424,40 @@ The <dfn>list of sufficiently strict speculative navigation referrer policies</d
       Otherwise it is uncredentialed, and redirects which would return to the original partition (thus ought to have credentials) will cause the prefetch to fail.
     </div>
 </div>
+
+<h2 id="sec-purpose-header">The `Sec-Purpose` HTTP request header</h2>
+
+The `` `Sec-Purpose` `` HTTP request header specifies that the request serves one or more purposes other than requesting the resource for immediate use by the user.
+
+The header field is an [[RFC8941]] Structured Header whose value must be a a <a spec="RFC8941">List</a>. Its ABNF is:
+
+```
+Sec-Purpose = sf-list
+```
+
+It may contain a <a spec="RFC8941">Item</a> member which is the <a spec="RFC8941">Token</a> "`prefetch`". If so, this indicates the request's purpose is to download a resource it is anticipated will be fetched shortly.
+
+<div class="issue">TODO: Are there normative implications of this that should be specified here?</div>
+
+The following parameters is defined for the "`prefetch`" member:
+
+* A parameter whose key is "`anonymous-client-ip`".
+
+  If present with a value other than boolean false, this parameter indicates that the prefetch request is being made using an anonymous client IP. Consequently, servers should not rely on it matching, or sharing a geographic location or network operator with, the client's IP address from which a non-prefetch request would have been made.
+
+  If a suitable response is not possible, for example because the resource depends on the client's geographic location, there is no other means of determining the location (e.g., the <a href="https://www.ietf.org/archive/id/draft-geohash-hint-00.html">Geohash client hint</a>), and no location-agnostic response is available, then the server should respond with an appropriate HTTP status code and response headers which mark the response as not suitable for caching.
```

Comment:
Do we have any idea what the appropriate status code would be? I could imagine 400, 403, 406.

On the other hand, maybe leaving this vague is better since it would avoid anyone depending on specific responses in some way.

Reply:
Yeah. I could also see 500, maybe. I don't want to overspecify and also we're kinda discussing in parallel whether prefetch discarding the response should be keyed off the status code or the cache control headers. (Presently at least Chromium actually does treat 4XX/5XX responses as potentially usable for prefetch.) If that settles down and the resolution seems interoperable I'm happy to narrow this.

Reply:
I think that for the purpose of prerendering it would be good to align on a single status code as "discard prerender" (I think 205 as in the prerender spec is the best option, to me 204 feels less descriptive). It would help guide server devs to be consistent by sending this status code rather than sending any 2xx/5xx and expecting the real navigation to re-fetch.